### PR TITLE
Refactoring Angular stuff for Mike

### PR DESCRIPTION
**Pull request by mlangens**

### CODE REVIEW

**Comment by mlangens** [comment]
@zaz @zazaian Hey Mike, I decided to go with CoffeeScript classes, because that seems to be the only way to manage dependencies across separate files. Please advise on any potential oversights I might have had here. If this looks good, I'd vote that we merge this in ASAP, then write unit tests so we don't end up stepping on each other's toes as we work on the same stuff. What do you think?

File: app/assets/javascripts/angular/prototypes/MetricPrototype.js.coffee [hunk 1]
```diff
@@ -1,5 +1,210 @@
...
+    this.addTiers(attrs["tiers"]) if attrs["tiers"] #add tiers if passed on init
+
+  alert: ()->
+    alert("snakes!")
+  addTier: (attrs={})->
+    self = this
+    newTier = new TierPrototype(self, attrs)
+    this.tiers.splice(-1, 0, newTier)
+
+    #???
```
Comment (mlangens) [inline app/assets/javascripts/angular/prototypes/MetricPrototype.js.coffee]: @zaz @zazaian I was unsure of what to include here between the two copies

**Comment by chcholman** [comment]
Testing through this branch: 

User story: As an instructor I need to be able to create new rubrics
- I click either of the "+ Criterion" buttons, and no new criterion appears
[blocked from testing further]

User story: As an instructor I need to be able to update an existing rubric
- I click "Edit Rubric" to enter the design mode, and no metrics/tiers appear (I've confirmed that they exist by viewing the rubric in another branch)
[blocked from testing further]

User story: As an instructor I need to be able to grade students using a rubric
- I click "Grade" to enter the grade mode, no grading items appear. 
[blocked from testing further]

@mlangens \ @zaz will one of you investigate?

**Comment by chcholman** [comment]
This looks great! @zaz will you review and confirm?

**Pull request merged**